pydantic_ai_slim/pydantic_ai/toolsets/processed.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from collections.abc import Awaitable
+from dataclasses import dataclass
+from functools import partial
+from typing import Any, Callable, Protocol
+
+from .._run_context import AgentDepsT, RunContext
+from .run import RunToolset
+from .wrapper import WrapperToolset
+
+
+class CallToolFunc(Protocol):
+    """A function protocol that represents a tool call."""
+
+    def __call__(self, name: str, tool_args: dict[str, Any], *args: Any, **kwargs: Any) -> Awaitable[Any]: ...
+
+
+ToolProcessFunc = Callable[
+    [
+        RunContext[AgentDepsT],
+        CallToolFunc,
+        str,
+        dict[str, Any],
+    ],
+    Awaitable[Any],
+]
+
+
+@dataclass
+class ProcessedToolset(WrapperToolset[AgentDepsT]):
+    """A toolset that lets the tool call arguments and return value be customized using a process function."""
```

Comment:
A process function???

Reply:
A function that wraps the underlying tool call and can modify arguments and its return value. I agree "process function" is more confusing than useful here, it can just be "function" as the rest of the docstring already explains what that function can do.